addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/D5_00/D5_00_01.java

```diff
@@ -36,11 +36,7 @@ public D5_00_01(ERP1Message packet) {
 
     @Override
     protected State convertToStateImpl(String channelId, String channelTypeId, State currentState, Configuration config) {
-        if (!isValid()) {
-            return UnDefType.UNDEF;
-        }
-
-        if (channelId == CHANNEL_CONTACT) {
+        if (channelId.equals(CHANNEL_CONTACT)) {
```

Comment:
Is the removal of the `!isValid()` check necessary? 
If I check the linked issue it's only suggested to replace `==` with `.equals(..)`.
It's also better to always put constants (`CHANNEL_CONTACT`) on the left hand side when using `.equals` to prevent NPEs.

Reply:
Hi @wborn, thanks a lot for your detailled review. The ``ìsValid()``` check can be removed, as it is done in the handler now. I will create a follow up issue to clean up all remaining EEP classes where this check is still done.
I fully agree with your second point regarding the comparison order. However this channelId is retrieved from a Channel where it is marked as ```@NonNull```. I will create another issue to handle this, as this comparison order is widely used in my binding.

Reply:
Thanks for the explanation @fruggy83. I now see the check in the handler and agree it's better cleanup all the EEP classes at once in another PR.